### PR TITLE
Map fix - beyond pivot

### PR DESCRIPTION
**Pull request by tzufgoogle**
Fix beyond pivot. Allow further POI on the path to be the beyond pivot.

### CODE REVIEW

File: cabby/geo/walk.py [hunk 1]
```diff
@@ -322,7 +319,7 @@ def get_landmark_if_tag_exists(self,
     return None
 
 
-  def pick_prominent_pivot(self, df_pivots: GeoDataFrame, end_point: Dict
+  def pick_prominent_pivot(self, df_pivots: GeoDataFrame, end_point: Dict, small: bool = False
   ) -> Optional[GeoSeries]:
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:323]: Please break this up over multiple lines. Also, please provided types for the keys and values for the Dict.

File: cabby/geo/walk.py [hunk 2]
```diff
@@ -457,31 +458,43 @@ def get_pivot_beyond_goal(self,
...
-      return GeoDataFrame(index=[0], columns=route.columns).iloc[0]
-
-    last_line_max = last_line[last_line['length'] ==
-                  last_line['length'].max()].iloc[0]
-
-    poly = last_line_max['geometry'].buffer(0.0001)
+    street_nodes = self.map.edges[condition_street_id]['u'].unique()
+    street_nodes = np.random.choice(street_nodes, MAX_NUM_BEYOND_TRY)
+    for i in range(MAX_NUM_BEYOND_TRY):
+      length = nx.shortest_path_length(self.map.nx_graph, source=street_nodes[i],target=last_node_in_route['osmid'])
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:466]: Can you reformat this to use multiple lines?

File: cabby/geo/walk.py [hunk 3]
```diff
@@ -457,31 +458,43 @@ def get_pivot_beyond_goal(self,
...
+      if length>3 and length<10:
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:467]: Please add a comment briefly explaining these threshold choices.

File: cabby/geo/walk.py [hunk 4]
```diff
@@ -457,31 +458,43 @@ def get_pivot_beyond_goal(self,
...
+        continue
+      path = nx.shortest_path(self.map.nx_graph, source=street_nodes[i], target=last_node_in_route['osmid'])
+      path.remove(last_node_in_route['osmid'])
+      if final_node_in_route['osmid'] in path:
+        path.remove(final_node_in_route['osmid'])
+
+      intersections = set(route['osmid']).intersection(path)
+      if len(intersections)<1 and len(path)>2:
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:475]: A brief comment explaining the intent of this if statement would help!

File: cabby/geo/walk.py [hunk 5]
```diff
@@ -457,31 +458,43 @@ def get_pivot_beyond_goal(self,
...
+  def select_pivot_from_path(self,
+                        route: GeoDataFrame,
+                        end_point: Dict,
+                        path: list):
+
+    path_nodes = self.map.nodes[self.map.nodes['osmid'].isin(path)]
+    points_route = path_nodes['geometry'].tolist()
+    poly = LineString(points_route).buffer(PIVOT_ALONG_ROUTE_MAX_DIST)
+    route_shape = LineString(route['geometry'].tolist()).buffer(PIVOT_ALONG_ROUTE_MAX_DIST)
+    poly = poly.difference(route_shape)
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:491]: Instead of reusing 'poly', please use a new variable name that documents what the new 'poly' is.

**Review by jasonbaldridge — COMMENTED** [review]
Just a few small formatting and comment requests!

**Review by tzufgoogle — COMMENTED** [review]
1. Made small changes in the logic of beyond pivot - instead of removing large POI any POI whether big or small is not picked if intersects the route taken.
2. Added comments to explain the logic
3. Broke the code into multiple lines

**Review by jasonbaldridge — COMMENTED** [review]
I'm not seeing the changes -- did you forget to push the code?

**Review by jasonbaldridge — APPROVED** [review]
LGTM

**Pull request merged**